experimental/dds/tree2/src/feature-libraries/editable-tree/editableField.ts

```diff
@@ -184,6 +186,49 @@ export class FieldProxyTarget extends ProxyTarget<FieldAnchor> implements Editab
 		this.context.insertNodes(fieldPath, index, newContent);
 	}
 
+	public moveNodes(
+		sourceIndex: number,
+		count: number,
+		destinationIndex: number,
+		destinationField?: EditableField,
+	): void {
+		const sourceFieldPath = this.cursor.getFieldPath();
+		const destinationFieldKindIdentifier =
+			destinationField !== undefined
+				? destinationField.fieldSchema.kind.identifier
+				: this.fieldSchema.kind.identifier;
+
+		assert(
+			this.fieldSchema.kind.identifier === sequence.identifier &&
+				destinationFieldKindIdentifier === sequence.identifier,
+			"Both source and destination fields must be sequence fields.",
+		);
+
+		assert(
+			keyIsValidIndex(destinationIndex, this.length + 1),
```

Comment:
I think this check is wrong. Shouldn't it be:
```suggestion
			destinationField === undefined || keyIsValidIndex(destinationIndex, destinationField.length + 1),
```

Reply:
moved the assert statement down and added `destinationFieldProxy === undefined || ` to the assert